### DISCUSSION TITLE
Unable to connect to devices using Resolvable Private Addresses

### DISCUSSION THREAD

**Discussion by shawnc722** [hbldh/bleak#1246] [Q&A]
* bleak version: 0.19.5
* Python version: 3.8
* Operating System: Linux and Windows
* BlueZ version (`bluetoothctl -v`) in case of Linux: 5.53

### Description

Pairing a device using a Resolvable Private Address works perfectly through the normal process, and then the device MAC is accessible. Using this MAC address to connect to the device in Bleak fails because a device with that address isn't found - it only sees the private address. The device name is also not resolved, so I can't find a way to connect to the device short of hardcoding a private address that changes every 15 minutes.
The behaviour I expected was that it would work similarly to bluetoothctl on linux, where it can connect to the proper MAC address.

### What I Did
Pair an iPhone or Android device to the computer running a recent enough version to have bluetooth privacy enabled, and then try to access that device from bleak using `BleakClient.connect(address)`.
The device isn't found because the address it's broadcasting doesn't match the true MAC address.

I'm not sure if I'm just overlooking something, but I can't see any way to connect to the device.

**Comment by dlech** [comment]
Generally, the way it works is that you scan with `BleakScanner` and look at the received advertising data to see if this look like the device I want to connect to. If it is, you pass the corresponding `BLEDevice` to the `BleakClient` constructor. The `detection_callback.py` example is an easy way to see what advertising data is available.
